components/create-post-page/IsbnEntry.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  Box,
+  Button,
+  HStack,
+  Icon,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+  useToast,
+  VStack,
+} from "@chakra-ui/react";
+import { PopulatedBook } from "@lib/services/book";
+import axios from "axios";
+import { FormEvent, useState } from "react";
+import { useQuery } from "react-query";
+import BarcodeIcon from "./BarcodeIcon";
+
+interface Props {
+  onIsbnSubmitted: (book: PopulatedBook) => void;
+}
+
+export default function IsbnEntry({ onIsbnSubmitted }: Props) {
+  const [isbn, setIsbn] = useState("");
+  const toast = useToast();
+
+  const bookQuery = useQuery("book-to-sell", () => {
+    return axios.get<PopulatedBook>(`/api/book/${isbn}/`).then((res) => res.data);
+  }, {
```

Comment:
Can we use this? https://github.com/calebmech/Book-Bazar/blob/33156c6fdec274be1587b74c1b65d37bf617e874/lib/hooks/book.ts#L5-L12

Reply:
I tried, although the implementation for making sure the loading icon works will get messy...

Reply:
@calebmech what do you think about the ISBN submit? It's pretty borked

Reply:
@calebmech ?